util/failuredomains/failure_domains_test.go

```diff
@@ -172,12 +179,696 @@ func TestNewFailureDomainPickMost(t *testing.T) {
+	machineB1New := &clusterv1.Machine{ObjectMeta: metav1.ObjectMeta{Name: "b1-new"}, Spec: clusterv1.MachineSpec{FailureDomain: ptr.To(b)}}
+	machineC1New := &clusterv1.Machine{ObjectMeta: metav1.ObjectMeta{Name: "c1-new"}, Spec: clusterv1.MachineSpec{FailureDomain: ptr.To(c)}}
+
+	testcases := []struct {
+		name             string
+		fds              clusterv1.FailureDomains
+		allMachines      collections.Machines
+		upToDateMachines collections.Machines
+		expected         []string
+	}{
+		// Use case A: 3 failure domains, 3 control plane machines
+
+		// scenario A.1: scale up to 3
+		{
+			name:             "3 failure domains, 0 machine, scale down from 0 to 1",
+			fds:              fds3,
+			allMachines:      collections.FromMachines(),
+			upToDateMachines: collections.FromMachines(),
+			expected:         []string{a, b, c}, // select fd a, b or c because they all have no up-to-date machines
+		},
+		{
+			name:             "3 failure domains, 1 machine, scale down from 1 to 2",
+			fds:              fds3,
+			allMachines:      collections.FromMachines(machineA1),
+			upToDateMachines: collections.FromMachines(machineA1),
+			expected:         []string{b, c}, // select fd b or c because they all have no up-to-date machines
+		},
+		{
+			name:             "3 failure domains, 2 machine, scale down from 2 to 3",
+			fds:              fds3,
+			allMachines:      collections.FromMachines(machineA1, machineB1),
+			upToDateMachines: collections.FromMachines(machineA1, machineB1),
+			expected:         []string{c}, // select fd c because it has no up-to-date machines
+		},
+
+		// scenario A.2: scale up during a rollout
+		{
+			name:             "3 failure domains, 3 outdated, scale up to 4",
+			fds:              fds3,
+			allMachines:      collections.FromMachines(machineA1Old, machineB1Old, machineC1Old),
+			upToDateMachines: collections.FromMachines(),
+			expected:         []string{a, b, c}, // select fd a, b or c because they all have no up-to-date machines, 1 machine overall
+		},
+		{
+			name:             "3 failure domains, 2 outdated, 1 new, scale up to 4",
+			fds:              fds3,
+			allMachines:      collections.FromMachines(machineA1New, machineB1Old, machineC1Old),
+			upToDateMachines: collections.FromMachines(machineA1New),
+			expected:         []string{b, c}, // select fd b or c because they all have no up-to-date machines (less than a)
+		},
+		{
+			name:             "3 failure domains, 1 outdated, 2 new, scale up to 4",
+			fds:              fds3,
+			allMachines:      collections.FromMachines(machineA1New, machineB1Old, machineC1Old),
+			upToDateMachines: collections.FromMachines(machineA1New, machineB1New),
+			expected:         []string{c}, // select fd c because it has no up-to-date machines (less than b, c)
+		},
+
+		// scenario A.3: scale up after machines has been remediated or forcefully deleted
+		{
+			name:             "3 failure domains, 3 machine, 1 new, scale up to 4",
```

Comment:
```suggestion
			name:             "3 failure domains, 2 outdated, 1 new, 1 empty FD, scale up to 4",
```

The format here is different than in above tests

Reply:
Fixed (those are machines, not FD)